mne/io/__init__.py

```diff
@@ -41,6 +41,7 @@
 from .nicolet import read_raw_nicolet
 from .artemis123 import read_raw_artemis123
 from .eeglab import read_raw_eeglab, read_epochs_eeglab, read_events_eeglab
+from .eeglab import read_annotations_eeglab
```

Comment:
why do we need a public facing API? Shouldn't it be enough to read it when you read the `raw` files?

Reply:
the vision is the following. For file format that have events with string descriptions which are for us Annotations we should read them as Annotations and then have a common function to go from Annotations to events. For now it's a mess. eeglab has its own internal mechanism which is different from EDF etc.

To summarize for EDF, EEGLAB, Brainvision the reader should read the annotations and not generate any stim channel. Then we should have a way to go from Annotations to events just by looking at annotations.onset (using orig_time) or to generate a stim_channel (just for preloaded data). It simplify massively the EEG data readers and make them more efficient as it would remove some allocations just to add a stim_channel.